packages/chain-agnostic-permission/src/caip25Permission.ts

```diff
@@ -483,3 +494,42 @@ function removeScope(
     operation: CaveatMutatorOperation.RevokePermission,
   };
 }
+
+/**
+ * Modifies the requested CAIP-25 permissions object after UI confirmation.
+ *
+ * @param caip25CaveatValue - The requested CAIP-25 caveat value to modify.
+ * @param accountAddresses - The list of permitted eth addresses.
+ * @param chainIds - The list of permitted eth chainIds.
+ * @returns The updated CAIP-25 caveat value with the permitted accounts and chainIds set.
+ */
+export const generateCaip25Caveat = (
+  caip25CaveatValue: Caip25CaveatValue,
```

Comment:
thoughts on `Omit<Caip25CaveatValue, 'requiredScopes' | 'optionalScopes'>` to make it clear that those values are going to get blown away?

Reply:
🤔 isn't it still expected that these properties will still be present in the caveat argument?

Reply:
Like this these values are expected (at the very least as empty objects) in the the calls to other helpers within this method: https://github.com/MetaMask/core/pull/5536/files#diff-8460bdba855631d866205e7540ef3ac2b612e97ac7698f2be49d2d586243b06cR222

Reply:
my bad. brain not working